x-pack/plugins/code/server/lsp/workspace_handler.ts

```diff
@@ -382,4 +385,17 @@ export class WorkspaceHandler {
     const workspaceRelativePath = path.relative(this.workspacePath, workspaceRepo.workdir());
     this.revisionMap[workspaceRelativePath] = headCommit.sha().substring(0, 7);
   }
+
+  private async checkFile(workspaceRepo: Repository, file: string) {
+    const headCommit = await workspaceRepo.getHeadCommit();
+    const entry = await headCommit.getEntry(file);
+    switch (entry.filemode()) {
+      case TreeEntry.FILEMODE.TREE:
+      case TreeEntry.FILEMODE.BLOB:
+      case TreeEntry.FILEMODE.EXECUTABLE:
```

Comment:
also, is it intended to disable the access to `TreeEntry.FILEMODE.LINK`? I am asking because we currently could render link file in the frontend. just a reminder.

Reply:
We could render but should not read the content, right? So here I think we should prevent lang-server from accessing to a link which points to an external file.

Reply:
make sense.